### ISSUE TITLE
How to compute Hunter/Toolchain/Config ID

### ISSUE THREAD

**Issue by chfast** [cpp-pm/hunter#658]
Is there a way to compute the Hunter/Toolchain/Config IDs without running full CMake configuration? I want to use these in CI deps cache ID.

**Comment by NeroBurner** [comment]
Yes, there is, but you'd have to reiimplement many of the things CMake does for you and the things Hunter does. If you want to try anyways, the following documentation could be very helpful for you

https://hunter.readthedocs.io/en/latest/overview/customization.html

**Comment by chfast** [comment]
Is there a way to run cmake to compute it by hunter but without downloading and building dependencies?

**Comment by NeroBurner** [comment]
I haven't tried, but you could create a CMakeLists.txt (or modify your project one), which exits right after the `HunterGate()` command and writes the Hunter-ID variables to a file, or just print them to the console)

What exactly would be the use-case? If you want a specific folder to be cached you could set a `HUNTER_ROOT` environment variable for each architecture you want to build or something like that. I'm doing that to get distinct hunter-cache folders for the different Ubuntu versions I build for (and then I can selectively delete them)

**Comment by chfast** [comment]
This should be a good cache key on CI servers.
